domain/machine/service/service_test.go

```diff
@@ -763,3 +768,98 @@ func (s *serviceSuite) TestSetLXDProfilesError(c *gc.C) {
 	err := NewService(s.state).SetAppliedLXDProfileNames(context.Background(), "666", []string{"profile1", "profile2"})
 	c.Check(err, jc.ErrorIs, rErr)
 }
+
+// TestSetReportedMachineAgentVersionInvalid is here to assert that if pass a
+// junk agent binary version to [Service.SetReportedMachineAgentVersion] we get
+// back an error that satisfies [coreerrors.NotValid].
+func (s *serviceSuite) TestSetReportedMachineAgentVersionInvalid(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	err := NewService(s.state).SetReportedMachineAgentVersion(
+		context.Background(),
+		cmachine.Name("0"),
+		coreagentbinary.Version{
+			Number: version.Zero,
+		},
+	)
+	c.Check(err, jc.ErrorIs, coreerrors.NotValid)
+}
+
+// TestSetReportedMachineAgentVersionSuccess asserts that if try to set the
+// reported agent version for a machine that doesn't exist we get an error
+// satisfying [machineerrors.MachineNotFound]
+func (s *serviceSuite) TestSetReportedMachineAgentVersionNotFound(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	// MachineNotFound error location 1
+	s.state.EXPECT().GetMachineUUID(gomock.Any(), cmachine.Name("0")).Return(
+		"", machineerrors.MachineNotFound,
+	)
+
+	err := NewService(s.state).SetReportedMachineAgentVersion(
+		context.Background(),
+		cmachine.Name("0"),
+		coreagentbinary.Version{
+			Number: version.MustParse("1.2.3"),
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Check(err, jc.ErrorIs, machineerrors.MachineNotFound)
+
+	machineUUID, err := uuid.NewUUID()
```

Comment:
It'd be good to split this into two tests.

Reply:
I have added a comment and explaining both locations in the test description.